src/main/java/com/cloudbees/hudson/plugins/folder/icons/BuildStatusFolderIcon.java

```diff
@@ -0,0 +1,84 @@
+package com.cloudbees.hudson.plugins.folder.icons;
+
+import com.cloudbees.hudson.plugins.folder.Folder;
+import com.cloudbees.hudson.plugins.folder.FolderIcon;
+import com.cloudbees.hudson.plugins.folder.FolderIconDescriptor;
+import hudson.Extension;
+import hudson.model.*;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.Stapler;
+
+public class BuildStatusFolderIcon extends FolderIcon {
+
+    Folder folder;
+
+    @DataBoundConstructor
+    public BuildStatusFolderIcon() {
+    }
+
+    public Integer ordinal() {
+      return this.getCombinedBallColor().ordinal();
+    }
+
+    private BallColor getCombinedBallColor() {
+        Boolean taskIsBuilding = false;
+        Boolean oneTaskIsIncomplete = false;
+        Result combinedResult = null;
+        for(Job job : this.folder.getAllJobs()) {
```

Comment:
Add a `// TODO switch to Items.allItems(this.folder, Job.class) once baseline Jenkins 2.37+` so we can pick up the perf update when we have https://github.com/jenkinsci/jenkins/blob/98ad8e4d73371bdcdca2f86ef74d92a5da339fc6/core/src/main/java/hudson/model/Items.java#L435 available